pkg/controller/metrics.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+const controllerNamespace = "controller"
+
+var (
+	ReconciliationDurations = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
```

Comment:
I like histogram to give a good feeling for getting a feeling for reconciliation time distributions. It might also be nice to have something which indicates how long the "current" loop has been running if its running. Together this would help determine when you can begin to suspect that a loop is hung.